### PR TITLE
[BACKLOG-6961] Fixed setting renderMode for callbacks.

### PR DESCRIPTION
**Pull request by annapopova1**
1. Fixed wrong behavior with render mode.
Steps:
 
 - open Inventory List report
 - uncheck Auto-Submit
 - select Line -> Motorcycles
 - see query in console. Should be request with renderMode: PARAMETER but now we have renderMode: XML
 
2. Also removed unnecessary condition for inSchedulerDialog.

@pentaho/orlandocalisbon please review and merge

### CODE REVIEW

File: package-res/reportviewer/reportviewer-prompt.js [hunk 1]
```diff
@@ -94,8 +94,6 @@ define(['common-ui/util/util', 'pentaho/common/Messages', "dijit/registry", 'com
             if (initFlag) {
               this._createPromptPanelFetchCallback(paramDefn);
```
Comment (jvelasques) [inline package-res/reportviewer/reportviewer-prompt.js]: @pentaho/orlandocalisbon could we remove the need for this initFlag, using something like [_.once](http://underscorejs.org/#once) ?

Comment (annapopova1) [inline package-res/reportviewer/reportviewer-prompt.js]: probably need to try this.. I have not familiarized with _.once

**Comment by annapopova1** [comment]
@pentaho/orlandocalisbon Applied _.once instead of init flag (thanks @jvelasques).

**Comment by jvelasques** [comment]
@pentaho/orlandocalisbon looks good to me

**Pull request merged**